native/membrane_video_compositor_common/src/plugins/transformation.rs

```diff
@@ -0,0 +1,72 @@
+use std::{any::Any, sync::Arc};
+
+use crate::WgpuContext;
+
+// Question to the reviewers: should the example below be in the finished documentation? Should it be
+//                            kept until the end of the plugins PR?
+//
+// Question to the reviewers, but for later: There are a couple of syntax sugar possibilities for the initialization described below.
+//                                           We could have a proc macro, a macro_rules macro, or possibly even leave it as-is but
+//                                           generate the boilerplate. Which one should we use?
```

Comment:
I would say that `macro_rules` macro would be sufficient so that not to repeat the implementations

Reply:
we'll need to discuss this a bit later, when it's time for introducing sugaring. I was more on the side of generating the boilerplate -- a mix task like `mix compositor.gen transformation`, which would generate a file with two functions you need to fill in yourself. Don't know if this would be possible yet though.

Reply:
If VC will be used only with Elixir, I would rather opt for creating a mix task. 
If we plan to integrate it with sth else in the future, the macro is probably a better option. 

I think this is outside of the scope of this task/PR.